test/problematic_tests/save_genome_test.py

```diff
@@ -142,29 +150,46 @@ def start_test(self):
         testname = inspect.stack()[1][3]
         print(('\n*** starting test: ' + testname + ' **'))
 
-    def fail_save_one_genome(self, params, error, exception=ValueError, contains=False):
+    def fail_save_genome(self, params, error, exception=ValueError, contains=False, mass=False):
         with self.assertRaises(exception) as context:
-            self.getImpl().save_one_genome(self.ctx, params)
+            if mass:
+                self.genome_interface.save_genome_mass(params)
+            else:
+                self.getImpl().save_one_genome(self.ctx, params)
         if contains:
             self.assertIn(error, str(context.exception))
         else:
             self.assertEqual(error, str(context.exception))
 
-    def check_save_one_genome_output(self, ret, genome_name):
+    def check_save_one_genome_output(
```

Comment:
This checker barely does checks anything, but I realize it was that way when you got here

Reply:
I could've sworn you spend a bunch of time adding rigorous tests to this module

Reply:
https://github.com/kbaseapps/GenomeFileUtil/pull/212#discussion_r1826117290

Reply:
This is similar to the comment below - the tests use the check_save_one_genome_output method as a helper for testing mass saves, so the mass save method isn't actually getting tested if it relies on that funtion

Reply:
Mass saves use the check_save_one_genome_output method as a helper. Why do you think the mass isn't actually being tested? In my opinion, it conducted some testing, but it wasn't tested thoroughly.

Reply:
Because check_save_one_genome output does very little testing - all it checks is the genome name, the type, and user name. There's test of the object contents, the attached files, the provenance, any of the other information in the object_info, etc.

Reply:
> There's no testing of the object contents, the attached files, the provenance, any of the other information in the object_info, etc.

I agree with you on this, but I thought we decided to document it for future work unless you changed your mind.

Reply:
Oh, I see, I misread this comment when I was on my phone: https://github.com/kbaseapps/GenomeFileUtil/pull/212#discussion_r1825252811

I thought it was saying tests needed to be added for the save_one_genome api call.

My position is that for the mass call that's been added to the API we should add tests to any code we change. Tests for single genome calls needed to be added, but for now can just be documented as needed. Where I'm confused is that for the mass call we're using the check_save_one_genome_output which hardly does anything, which therefore means we're not testing the mass call - or we're calling that function for reasons I don't understand, since it has almost no benefit

Reply:
The mass call is not added to the API; save_genome_mass is used internally.

Reply:
I'm talking about the mass call that was added as a part of this entire parllelization project